src/web/demo.rkt

```diff
@@ -228,83 +242,33 @@
-    (for ([page (all-pages result)])
-      (call-with-output-file (build-path (*demo-output*) path page)
-        (λ (out) 
-          (with-handlers ([exn:fail? (page-error-handler result page out)])
-            (make-page page out result (*demo-output*) #f)))))
-    (update-report result path seed
-                    (build-path (*demo-output*) "results.json")
-                    (build-path (*demo-output*) "index.html"))))
-
-; A place holder helper function that should be used when no work needs to be
-; done on the finished result
-(define (default-after result job-id seed) empty)
+  (hash-set! *completed-jobs* job-id result))
 
 (define (run-job job-info)
- (define sema (work-sema job-info))
- (define job-id (work-id job-info))
+ (match-define (work job-id info sema) job-info)
  (define path (format "~a.~a" job-id *herbie-commit*))
  (cond ;; Check caches if job as already been completed
   [(hash-has-key? *completed-jobs* job-id)
-  (semaphore-post sema)]
+   (semaphore-post sema)]
   [(and (*demo-output*) (directory-exists? (build-path (*demo-output*) path)))
-  (semaphore-post sema)]
-  [else (match (work-info job-info)
-    [(list 'improve formula)
-      (wrapper-run-herbie 
-       (run-herbie-command 'improve formula (get-seed) #f #f #f) 
-       job-id after-improve)]
-    [(list 'sample formula seed*)
-      (wrapper-run-herbie 
-       (run-herbie-command 'sample formula seed* #f #f #t) 
-       job-id default-after)]
-    [(list 'errors formula seed* pcontext)
-      (wrapper-run-herbie 
-       (run-herbie-command 'errors formula seed* pcontext #f #t) 
-       job-id default-after)]
-    [(list 'exacts formula seed* pcontext)
-      (wrapper-run-herbie 
-       (run-herbie-command 'exacts formula seed* pcontext #f #t) 
-       job-id default-after)]
-    [(list 'evaluate formula seed* pcontext)
-      (wrapper-run-herbie 
-       (run-herbie-command 'evaluate formula seed* pcontext #f #t) 
-       job-id default-after)]
-    [(list 'local-error formula seed* pcontext)
-      (wrapper-run-herbie 
-       (run-herbie-command 'local-error formula seed* pcontext #f #t) 
-       job-id default-after)]
-    [(list 'alternatives formula seed* pcontext)
-      (wrapper-run-herbie 
-       (run-herbie-command 'alternatives formula seed* pcontext #f #t) 
-       job-id default-after)]
-    [(list 'cost formula)
-      (wrapper-run-herbie 
-       (run-herbie-command 'cost formula #f #f #f #t) 
-       job-id default-after)])])
- (eprintf "Job ~a complete\n" job-id)
- (hash-remove! *jobs* job-id)
- (semaphore-post sema))
+   (semaphore-post sema)]
+  [else (wrapper-run-herbie info job-id)
+   (eprintf "Job ~a complete\n" job-id)
+   (hash-remove! *jobs* job-id)
+   (semaphore-post sema)]))
 
 ; Handles semaphore and async part of a job
-(struct work (id info sema))
+(struct work (id job sema))
 
 ; Encapsulates semaphores and async part of jobs.
-(define (run-work sync-job job-id job)
+(define (run-work #:sync? [sync-job? #f] job)
```

Comment:
Probably give this a default of `#t` since that's the more common option, and then you can avoid mentioning it everywhere.